tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,243 @@
+    """
+    is_running = is_container_running(duthost, container_name)
+    if is_running == "false":
+        pytest_assert(False, "{} is not running".format(container_name))
+
+    if is_process_name:
+        kill_process_by_name(duthost, container_name, process_name)
+    else:
+        kill_process_by_pid(duthost, container_name, program_name, program_pid)
+
+    logging.info("Waiting until {} is stopped...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, True), 
+                  "Failed to stop {}".format(container_name))
+    logging.info("{} is stopped".format(container_name))
+
+    logging.info("Waiting until {} is restarted...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "Failed to restart {}".format(container_name))
+    logging.info("{} is restarted".format(container_name))
+
+def verify_no_autorestart_with_non_critical_process(duthost, container_name, process_name):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    is_running = is_container_running(duthost, container_name)
+    if is_running == "false":
+        pytest_assert(False, "{} is not running".format(container_name))
+
+    kill_process_by_name(duthost, container_name, process_name)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "{} is stopped unexpectedly".format(container_name))
+    logging.info("{} is running".format(container_name))
+
+def test_containers_autorestart(duthost):
+    container_autorestart_info = get_autorestart_container_and_state(duthost)
+    for container_name in container_autorestart_info:
+        if container_name in ["restapi"]:
+            continue
+        
+        logging.info("Change {} auto-restart state to 'enabled'".format(container_name))
+        duthost.shell("config container feature autorestart {} enabled".format(container_name))
+        
+        verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd")
+
+        if container_name in ["pmon", "radv"]:
+            logging.info("Restore {} auto-restart state to {}".format(container_name, container_autorestart_info[container_name]))
+            duthost.shell("config container feature autorestart {} {}".format(container_name, container_autorestart_info[container_name]))
+            continue
```

Comment:
Why are you doing this here for these two containers?

Reply:
Since there are no critical processes running in PMon and radv on virtual testbed, so I skipped the testing for these two containers.

Reply:
But these tests will also run on physical switches, so we will want to test these containers.

Reply:
Yes, I will remove this part.